src/connection-manager/dialer/index.ts

```diff
@@ -260,21 +260,28 @@ export class DefaultDialer implements Startable, Dialer {
       addrs.map(async (ma) => await this._resolve(ma, options))
     ))
       .flat()
-      // Multiaddrs not supported by the available transports will be filtered out.
-      .filter(ma => Boolean(this.components.transportManager.transportForMultiaddr(ma)))
+
+    // Multiaddrs not supported by the available transports will be filtered out.
+    addrs = addrs.filter(ma => Boolean(this.components.transportManager.transportForMultiaddr(ma)))
 
     // deduplicate addresses
-    addrs = [...new Set(addrs.map(ma => ma.toString()))].map(ma => multiaddr(ma))
+    let filteredAddrs = [...new Set(addrs.map(ma => ma.toString()))].map(ma => multiaddr(ma))
```

Comment:
This is a little bit difficult to read, could we have something like:

```typescript
const filteredAddrs = addrs.filter(ma => Boolean(this.components.transportManager.transportForMultiaddr(ma)))

const dedupedAddrs = [...new Set(filteredAddrs.map(ma => ma.toString()))]

const dedupedMultiaddrs = dedupedAddrs.map(ma => multiaddr(ma))
```

Reply:
Done!